[Docs] coupled solid documentation (#1539)
GEOSX/GEOSX: src/coreComponents/constitutive/docs/PorousSolids.rst

.. _PorousSolids:

############################################
Porous Solids
############################################

Overview
========================

> Simulation of fluid flow in poroous media and of poromechanics,

Reviewer: ```suggestion
Simulation of fluid flow in porous media and poromechanical coupling problems,
```

requires to define, along with fluid properties, the hydrodynamical properties of
the solid matrix. Thus, for porous media flow and and poromecanical simulation in GEOSX,
two types of composite constitutive models can be defined to specify the characteristics

> of a porous material: (1) a `CompressibleSolid` model, used for flow only simulations and which

Reviewer 2: ```suggestion
of a porous material: (1) a `CompressibleSolid` model, used for flow-only simulations and which
```

assumes that all poromechanical effects can be represented by the pressure dependency of the

> porosity; (2) a `PorousSolid` model which, instead allows to couple any solid model with

Reviewer 2: ```suggestion
porosity; (2) a `PorousSolid` model which, instead, allows to couple any solid model with
```

> a `BiotPorosity` model and to include permeability dependency on the mechanical response.

Reviewer 2: ```suggestion
a `BiotPorosity` model and to include permeability's dependence on the mechanical response.
```

> Both these composite  models require the names of the solid, porosity and permeability models

Reviewer: ```suggestion
Both composite  models require the names of the solid, porosity and permeability models
```

> Both these composite  models require to provide the names of the solid, porosity and permeability models

Reviewer 2: ```suggestion
Both these composite  models require the names of the solid, porosity and permeability models
```

that, combined, define the porous material. The following sections outline how these models can be
defined in the Constitutive block of the xml input files and which type of submodels they
allow for.

CompressibleSolid
========================

> This composite constitutive model requires to define a `NullModel` as solid model, since

Reviewer 2: ```suggestion
This composite constitutive model requires to define a `NullModel` as solid model (since
```

> no mechanical properties are used, a `PressurePorosity` model and any type of `Permeability` model.

Reviewer 2: ```suggestion
no mechanical properties are used), a `PressurePorosity` model and any type of `Permeability` model.
```

> To define this composite model the keywork `CompressibleSolid` has to be appendended to the name

Reviewer 2: ```suggestion
To define this composite model the keyword `CompressibleSolid` has to be appendend to the name
```

> of the peremability model of choice, as shown in the following example for the `ConstantPermeability` one.

Reviewer 2: ```suggestion
of the permeability model of choice, as shown in the following example for the `ConstantPermeability` model.
```

e.g., PorousElasticIsotropic, PorousDruckerPrager, and so on. Additionally, they require to
define a `BiotPorosity` model and a `ConstantPermeability` model. For example, a Poroelastic material
with a certain permeability can be defined as

.. code-block:: xml

    <Constitutive>
     <PorousElasticIsotropic name="porousRock"
                             porosityModelName="rockPorosity"
                             solidModelName="rockSkeleton"
                             permeabilityModelName="rockPermeability"/>

     <ElasticIsotropic name="rockSkeleton"
                       defaultDensity="0"
                       defaultYoungsModulus="1.0e4"
                       defaultPoissonRatio="0.2"/>

     <BiotPorosity name="rockPorosity"
                   grainBulkModulus="1.0e27"
                   defaultReferencePorosity="0.3"/>

  <ConstantPermeability name="rockPermeability"
                        permeabilityComponents="{ 1.0e-4, 1.0e-4, 1.0e-4 }"/>
    </Constitutive>

Note that any of the previously described solid models is used by the `PorousSolid` model
to compute the effective stress, leading to either poro-elastic, poro-plastic, or poro-damage

> behavior depending on the specific model chose.

Reviewer 2: ```suggestion
behavior depending on the specific model chosen.
```
